app/views/exams/_form.html.erb

```diff
@@ -1,20 +1,29 @@
-<%= form_for @exam do |f| %>
+<%= form_for @exam, html: {id: "form_exam_submit"} do |f| %>
   <ol>
-    <% @questions.each do |question| %>
+    <% @questions.includes(:answers).each do |question| %>
       <li>
         <strong><%= question.question_content %></strong>
           <%= f.fields_for :detail_question_answer do |detail_a| %>
             <% if question.single_choice? %>
                 <% question.answers.each do |answer| %>
                   <%= detail_a.label "#{question.id}_#{answer.id}", answer.answer_content do %>
+                    <% if  @user_answered[question.id]&.include?(answer.id.to_s) %>
+                    <%= detail_a.radio_button answer.question_id, answer.id, checked: true %>
+                    <% else %>
                     <%= detail_a.radio_button answer.question_id, answer.id %>
```

Comment:
indent và code này lặp nè... tìm cách code thành 1 line
gợi ý: dùng toán tử ngôi số 3 và tạ method check của if vào helper